appcues/src/main/java/com/appcues/ui/ExperienceRenderer.kt

```diff
@@ -107,15 +108,20 @@ internal class ExperienceRenderer(
             return true
         }
 
-        val success = show(qualifiedExperiences.first())
-        if (!success) {
+        qualifiedExperiences.filter { it.renderContext is RenderContext.Embed }.forEach {
+            // TODO what to do here
+            show(it)
+        }
+
+        val modalSuccess = show(qualifiedExperiences.first { it.renderContext is RenderContext.Modal })
+        if (!modalSuccess) {
             val remainingExperiences = qualifiedExperiences.drop(1)
             if (remainingExperiences.isNotEmpty()) {
                 // fallback logic - try the next remaining experience, if available
                 return show(remainingExperiences)
             }
         }
-        return success
+        return modalSuccess
```

Comment:
if I'm following correctly - we'll need to trim out the embed experiences before doing any sort of recursion on line 121 to try lower priority modal experiences - so the embeds don't get re-triggered on line 113?

Reply:
you are right here

Reply:
the show(experiences) method doesnt need to return true or false.

created two separate functions for embeds and modal, embeds will evolve later, modal is now doing what the original show was doing. 

trying to show a modal experience, if it fails it will drop that and call it again until the list is empty.